tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -25,6 +26,8 @@ namespace Datadog.Trace.Ci;
 public sealed class Test
 {
     private static readonly AsyncLocal<Test?> CurrentTest = new();
+    private static readonly HashSet<Test> OpenedTests = new();
```

Comment:
Just floating another idea, not sure it's better or anything, just food for thought:

- Use `ConcurrentDictionary` instead of `HashSet` so you don't need the `lock`